rundeckapp/grails-app/services/rundeck/services/DbStorageService.groovy

```diff
@@ -179,7 +179,14 @@ class DbStorageService implements NamespacedStorage{
             }else{
                 isNull('namespace')
             }
-            eq('dir', dir)
+            if(dir){
+                eq('dir', dir)
+            }else{
+                or{
+                    eq('dir', dir)
```

Comment:
to clarify that this is supposed to search empty string, I would make it explicit here
```suggestion
                    eq('dir','')
```